Make.versions

```diff
@@ -0,0 +1,47 @@
+#
+# A release branch requires updating the following two variables at the bottom of this file:
+#
+# IOS_PACKAGE_VERSION (major/minor/revision #)
+# MAC_PACKAGE_VERSION (major/minor/revision #)
+#
+# Update version numbers on master as well, to the next version
+#
+
+#
+# ** Version numbers **
+#
+# Versions are defined as: Major.Minor.Revison.Build
+#
+# Major/minor (first/second numbers - max 2 digits each):
+# - Bump for major/minor releases.
+#
+# Revision (third number - max 2 digits):
+# - Reset to 0 after a major or minor bump (do not use 99 for Xcode preview
+#   branches (use 0 instead), because otherwise we can't bump it further if
+#   needed).
+# - Bump for service releases and previews.
+# - Bump if commit distance becomes > 999.
+# - Can also be bumped for other reasons (in particular there's no correlation
+#   between Preview/Service Release #X and Revision #Y).
+#   - Bumping revision to a high enough number to make it clear that there's
+#     no correlation is a valid reason to bump.
+# - The revision must be bumped at the same time for both iOS and Mac
```

Comment:
So shouldn't we have one `rev` variable shared between iOS and Mac?

Reply:
they don't have to be identical, e.g. if we merge `xcodeX` then we might only change XI revision (back to 0) but keep a high(er) revision for XM

Reply:
Oh ok so if we bump XI back to 0 the XM rev needs to be +1.

That's important. IMHO the comment should be divided in 2:

- Explanations of the various numbers and would _can_ be done
- TODO: things to check when changing any numbers

Reply:
Cause that one line for instance doesn't jump out but every time we update the package version we should re-read it to not forget (:

Reply:
We needed this _quickly_ since 15.9 numbers got broken/reset with xcode10 merge (and missing `0`) but that should not stop adding more documentation in master. Please prepare a PR :)